smartlearner/batch_schedulers.py

```diff
@@ -47,6 +48,18 @@ def __iter__(self):
             self.shared_batch_count.set_value(batch_count)
             yield batch_count + 1
 
+    def save(self, savedir="./"):
+        state = {"version": 1,
```

Comment:
I found version better suited because on load we can do some check and adapt the code accordingly (e.g. version 1 was missing some attribute 'xyz' so we should add it whenever loading code from version <2.

But keeping the hash of the code is an interesting idea even though the hash will change as soon as we make small modification (e.g. spacing). I'll write a Issue so we can keep it in mind but for now I think manual version is more manageable.